docs/decisions/0063-vector-store-list-keys.md

```diff
@@ -0,0 +1,118 @@
+}
+```
+Pros:
+- This is a non-breaking change, each vector store record collection must choose to implement this interface.
+
+Cons:
+- This is a more complex way to add the functionality
+
+## Option 3: alter the behavior of the Get/GetBatch methods to allow getting all records (key=None or similar)
+
+This options alters the behavior of the Get methods to allow getting all records, this can be done by passing a special key value (e.g. None) to the Get method.
+
+```csharp
+
+public interface IVectorStoreRecordCollection<TKey, TRecord> : IVectorizedSearch<TRecord>
+{
+    Task<TRecord?> GetAsync(TKey? key, GetRecordOptions? options = default, CancellationToken cancellationToken = default);
+    IAsyncEnumerable<TRecord> GetBatchAsync(IEnumerable<TKey>? keys, GetRecordOptions? options = default, CancellationToken cancellationToken = default);
+}
+
+```
+Pros:
+- This allows the other options to be used in concert with getting all records
+- This is a non-breaking change
+Cons:
+- This returns the full records instead of just the keys, which can be useful in some cases, but can be overhead in others.
+- It is not immediately clear that passing a special key value will return all records, a dedicated method would be more clear.
+
+## Options 3b: alter the behavior of the Get/GetBatch methods to allow getting all keys
+
+This options alters the behavior of the Get methods to allow getting all keys, this can be done by passing a special key value (e.g. None) to the Get method, combined with a change in the GetRecordOptions, to specify Keys only.
+
+```csharp
+public interface IVectorStoreRecordCollection<TKey, TRecord> : IVectorizedSearch<TRecord>
+{
+    Task<TRecord?> GetAsync(TKey? key, GetRecordOptions? options = default, CancellationToken cancellationToken = default);
+    IAsyncEnumerable<TRecord> GetBatchAsync(IEnumerable<TKey>? keys, GetRecordOptions? options = default, CancellationToken cancellationToken = default);
+}
+public class GetRecordOptions
+{
+    public bool IncludeVectors { get; init; } = false;
+    public bool KeysOnly { get; init; } = false;
+}
+
+```
+Pros:
+- This allows the other options to be used in concert with getting all keys
+- This is a non-breaking change
+- This allows the user to choose if they want the full records or just the keys
+Cons:
+- This is a more complex way to add the functionality
+- It is not immediately clear that passing a special key value will return all records, a dedicated method would be more clear.
+
+## Naming of the method
+
+In the examples above the term ListKeys is used, but other terms could be used as well, such as GetKeys, Keys, etc. The term ListKeys is chosen because it is clear that this method will return a list of keys.
+
+## Decision
+Option 1: add a new method to the VectorStoreRecordCollection interface
+
+This is the cleanest approach, and for Python, the breaking change can be mitigated by providing a default implementation in the interface that returns an empty list.
```

Comment:
Before describing `Decision` section, let's discuss it with the team and then update this section based on team's agreement.